### DISCUSSION TITLE
(parking-v0) Is it possible (and how) to add a penalty when car crosses a line?

### DISCUSSION THREAD

**Discussion by mmarfat** [eleurent/highway-env#468] [General]
Hi,

I'm wondering whether it's possible to implement a penalty which will be applied whenever the car crosses the parking spot lines (any parking spot, not just the target one)? The idea is to encourage the agent to park "by the rules".

If there's a better way to encourage the agent not to cross the line please suggest as I'm completely new to RL and this is just my initial idea.

**Comment by eleurent** [comment]
Its possible to edit the parking env's reward function here:

https://github.com/Farama-Foundation/HighwayEnv/blob/21225dbe12a50c088540c0257bcc05b03a97732e/highway_env/envs/parking_env.py#L203

You can compare the vehicle's position and heading to each lane's and figure out if they are crossing, to add a penalty.

**Comment by eleurent** [comment]
Another possibility is maybe to increase the weight associated to the heading error, such that the agent prioritises being aligned with the parking spots rather than being in the correct location. They are the two 0.02 weights at

https://github.com/Farama-Foundation/HighwayEnv/blob/21225dbe12a50c088540c0257bcc05b03a97732e/highway_env/envs/parking_env.py#L90

**Comment by mmarfat** [comment]
Thanks, I will try it!
